interfaces/opensearch_client/v0/README.md

```diff
@@ -0,0 +1,88 @@
+
+## Direction
+
+```mermaid
+flowchart TD
+    Requirer -- index, \nextra-user-roles --> Provider
+    Provider -- username, \npassword, \nendpoints --> Requirer
+```
+
+As with all Juju relations, the `opensearch-client` interface consists of two parties: a Provider (opensearch charm), and a Requirer (application charm). The Requirer will be expected to provide an index name, and the Provider will provide new unique credentials (along with other optional fields), which can be used to access the index itself.
+
+## Behavior
+
+Both the Requirer and the Provider need to adhere to criteria to be considered compatible with the interface.
+
+### Provider
+
+- Is expected to create an application user inside the opensearch cluster when the requirer provides the `index` field.
+  - This user is removed when the relation is removed.
+  - Is expected to apply the permissions in the `extra-user-roles` provided by the Requirer to this user (e.g. `extra-user-roles=admin`).
+  - If multiple relations require the same index name, they should all be able to access it.
+- Is not expected to create an index on relation creation.
+  - Responsibility for managing an index rests with the requirer application, including creating and removing indices.
+- Is expected to provide unique `username` and `password` fields as Juju Secrets when Requirer provides the `index` field.
+- Is expected to provide the `endpoints` field containing all cluster endpoint addresses in a comma-separated list.
+- Is expected to provide the `version` field describing the installed version number of opensearch.
+- If the charm has TLS enabled (such as using the [TLS Certificates Operator](https://github.com/canonical/tls-certificates-operator)), it is expected to provide the CA chain in the `tls-ca` field as a Juju Secret.
+
+### Requirer
+
+- Is expected to provide an index name in the `index` field.
+- Is expected to manage its own index.
+  - Indices are not created on the provider application when the relation is created. The `index` field exists to grant the correct permissions for the relation user, which the requirer charm uses to control its index.
+  - This index is NOT removed from the provider charm when the relation is removed.
+- Is expected to have different relations with the same interface name if Requirer needs access to multiple opensearch indices.
+- Is expected to provide user permissions in the `extra-user-roles` field. These permissions will be applied to the user created for the relation.
+  - These permissions can be set to two values:
+    - default: this has read-write permissions over the index that has been generated for this relation. This permission level will be applied if no value is provided.
```

Comment:
Why is `default` an **extra** user role? Shouldn't it just be this one by default

Reply:
I've updated this to better explain that default is applied if no value is provided